WordPress/src/main/java/org/wordpress/android/ui/mysite/dynamiccards/DynamicCardsSource.kt

```diff
@@ -18,27 +18,40 @@ class DynamicCardsSource
 @Inject constructor(
     private val dynamicCardStore: DynamicCardStore,
     private val selectedSiteRepository: SelectedSiteRepository
-) : MySiteSource<DynamicCardsUpdate> {
-    private val refresh = MutableLiveData<Boolean>()
+) : MySiteRefreshSource<DynamicCardsUpdate> {
+    override val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
 
-    override fun buildSource(coroutineScope: CoroutineScope, siteLocalId: Int): LiveData<DynamicCardsUpdate> {
+    override fun build(coroutineScope: CoroutineScope, siteLocalId: Int): LiveData<DynamicCardsUpdate> {
         val data = MediatorLiveData<DynamicCardsUpdate>()
         data.refreshData(coroutineScope, siteLocalId)
         data.addSource(refresh) {
-            data.refreshData(coroutineScope, siteLocalId)
+            data.refreshData(coroutineScope, siteLocalId, refresh.value)
         }
         return data
     }
 
+    private fun MediatorLiveData<DynamicCardsUpdate>.refreshData(
+        coroutineScope: CoroutineScope,
+        siteId: Int,
+        isRefresh: Boolean? = null
+    ) {
+        when (isRefresh) {
+            null, true -> refreshData(coroutineScope, siteId)
+            false -> Unit // Do nothing
+        }
+    }
+
     private fun MediatorLiveData<DynamicCardsUpdate>.refreshData(coroutineScope: CoroutineScope, siteId: Int) {
         coroutineScope.launch {
             val cards = dynamicCardStore.getCards(siteId)
-            this@refreshData.postValue(
-                    DynamicCardsUpdate(
-                            pinnedDynamicCard = cards.pinnedItem,
-                            cards = cards.dynamicCardTypes
-                    )
-            )
+//            refresh.postValue(false)
+//            this@refreshData.postValue(
+//                    DynamicCardsUpdate(
+//                            pinnedDynamicCard = cards.pinnedItem,
+//                            cards = cards.dynamicCardTypes
+//                    )
+//            )
```

Comment:
Question (❓): I have the feeling that this commented out code is an oversight and can be removed, right?

Reply:
Oh yes, it is. Serves me right for not leaving a todo there for myself. 🤦